src/test/regress/sql/aggregate_support.sql

```diff
@@ -533,5 +534,44 @@ SELECT floor(AVG(COALESCE(agg_col, 10)))
 FROM dist_table
 LEFT JOIN ref_table ON TRUE;
 
+-- try createing aggregate having non-distributable dependency type
+create table dummy_tbl (a int);
+create function dummy_fnc(a dummy_tbl, d double precision) RETURNS dummy_tbl
+    AS $$SELECT 1;$$ LANGUAGE sql;
+-- should give warning and create aggregate local only
+create aggregate dependent_agg (float8) (stype=dummy_tbl, sfunc=dummy_fnc);
+
+-- clear and try again with distributed table
+DROP TABLE dummy_tbl CASCADE;
+
+create table dummy_tbl (a int);
+SELECT create_distributed_table('dummy_tbl','a');
+create function dummy_fnc(a dummy_tbl, d double precision) RETURNS dummy_tbl
+    AS $$SELECT 1;$$ LANGUAGE sql;
+
+-- test in tx block
+-- should create and distribute
+BEGIN;
+create aggregate dependent_agg (float8) (stype=dummy_tbl, sfunc=dummy_fnc);
+COMMIT;
+--verify
+SELECT run_command_on_workers($$select aggfnoid from pg_aggregate where aggfnoid::text like '%dependent_agg%';$$);
+
```

Comment:
Can you also check the pg_dist_object whether aggregates is marked as distributed on all nodes?